src/main/java/org/jabref/logic/crawler/StudyRepository.java

```diff
@@ -0,0 +1,344 @@
+            createQueryResultFolder(query);
+            converter.getActiveFetchers()
+                     .forEach(searchBasedFetcher -> createFetcherResultFile(query, searchBasedFetcher));
+            createQueryResultFile(query);
+        }
+        createStudyResultFile();
+    }
+
+    /**
+     * Creates a folder using the query and its corresponding query id.
+     * This folder name is unique for each query, as long as the query id in the study definition is unique for each query.
+     *
+     * @param query The query the folder is created for
+     */
+    private void createQueryResultFolder(String query) throws IOException {
+        Path queryResultFolder = getPathToQueryDirectory(query);
+        createFolder(queryResultFolder);
+    }
+
+    private void createFolder(Path folder) throws IOException {
+        if (Files.notExists(folder)) {
+            try {
+                Files.createDirectory(folder);
+            } catch (IOException e) {
+                throw new IOException("Error during creation of repository structure.", e);
+            }
+        }
+    }
+
+    private void createFetcherResultFile(String query, SearchBasedFetcher searchBasedFetcher) {
+        String fetcherName = searchBasedFetcher.getName();
+        Path fetcherResultFile = getPathToFetcherResultFile(query, fetcherName);
+        createBibFile(fetcherResultFile);
+    }
+
+    private void createQueryResultFile(String query) {
+        Path queryResultFile = getPathToFetcherResultFile(query, "result");
+        createBibFile(queryResultFile);
+    }
+
+    private void createStudyResultFile() {
+        createBibFile(getPathToStudyResultFile());
+    }
+
+    private void createBibFile(Path file) {
+        if (Files.notExists(file)) {
+            try {
+                Files.createFile(file);
+            } catch (IOException e) {
+                throw new IllegalStateException("Error during creation of repository structure.", e);
+            }
+        }
+    }
+
+    /**
+     * Returns a string that can be used as a folder name.
+     * This removes all characters from the query that are illegal for directory names.
+     * Structure: ID-trimmed query
+     *
+     * Examples:
+     * Input: '(title: test-title AND abstract: Test)' as a query entry with id 1
+     * Output: '1 - title= test-title AND abstract= Test'
+     *
+     * Input: 'abstract: Test*' as a query entry with id 1
+     * Output: '1 - abstract= Test'
+     *
+     * Input: '"test driven"' as a query entry with id 1
+     * Output: '1 - test driven'
+     *
+     * @param query that is trimmed and combined with its query id
+     * @return a unique folder name for any query.
+     */
+    private String trimNameAndAddID(String query) {
+        // Replace all field: with field= for folder name
+        String trimmedNamed = query.replaceAll(":", "=");
+        trimmedNamed = trimmedNamed.replaceAll("[^A-Za-z0-9_.\\s=-]", "");
```

Comment:
Extract the regex to a
private final static string Pattern  =  Pattern.compile(regex)
pattern.matcher(query).replaceAll("")